azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -67,7 +67,11 @@ private void restartExecutableFlow(final ExecutableFlow exFlow) {
     }
     options.setMailCreator(flow.getMailCreator());
     // Update the flow options so that the flow will be not retried again by Azkaban
-    options.setExecutionRetried(true);
+
+    // inherent the retry time counters
+    executableFlow.setUserDefinedRetryCount(exFlow.getUserDefinedRetryCount());
+    executableFlow.setSystemDefinedRetryCount(exFlow.getSystemDefinedRetryCount());
+
     // If a retried flow A gets retried again with a new execution id flow B, the original flow
     // execution id of flow B should be the same as flow A's original flow execution id.
     if (options.getOriginalFlowExecutionIdBeforeRetry() == null) {
```

Comment:
Wondering if we should track original execution id as an attribute of ExecutableFlow, just like counters. It would help keep related things together I think. What are the benefits of adding original exec id to ExecutionOptions?

Reply:
Hi yes it's in the plan, should be in ExecutableFlow, and will have a follow up PR to address this.